include/rocksdb/options.h

```diff
@@ -882,6 +882,23 @@ struct DBOptions {
   // Default: false
   bool enable_pipelined_write = false;
 
+  // If false, rocksdb does not advance the sequence number for new snapshots
+  // unless all the writes with lower sequence numbers are already finished.
+  // This provides the immutability that we except from snapshots. If true, then
+  // this property is relaxed: the writes issues after the snapshot is obtained
+  // (with larger sequence numbers) will be still not visible to the reads from
+  // that snapshot, however, there still might be pending writes (with lower
+  // sequence number) that will change the state visible to the snapshot after
+  // they are landed to the memtable. unordered_write trades higher write
+  // throughput with relaxing the guarantees of snapshots. If the application
+  // cannot tolerate the relaxed guarantees, it can implement its own mechanisms
+  // to work around that and yet benefit from the higher throughput. Using
+  // TransactionDB with WRITE_PREPARED write policy is one way to achieve
+  // immutable snapshots despite unordered_write.
```

Comment:
Discusssed offline.
I think we should discuss the behavior not just for snapshots, but also iterators, Get() and MultiGet() when being used without snapshots. If we introduces it, it will be good.

I suggest the contract is explained first, and then the description about the implementation, e.g. how sequence number is advanced, etc.